plugins/modules/wait_for_task.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2023, Julien Godin <julien.godin@camptocamp.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: wait_for_task
+version_added: 3.13.0
+short_description: Wait for a task
+description:
+  - Wait for a task to finish
+author:
+  - "Julien Godin (@JGodin-C2C)"
+options:
+  task:
+    description:
+      - Task id to wait for.
+    required: true
+    type: str
+  timeout:
+    description:
+      - How much time the task should take to be finished
+    required: false
+    type: int
+    default: 60
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+'''
+
+
+EXAMPLES = '''
+- name: wait for a task to finish in at least 10 seconds
+  theforeman.foreman.wait_for_task:
+    server_url:  "https://foreman.example.com"
+    password: changeme
+    username: admin
+    task: "{{ item }}"
+    timeout: 60
+  loop: "{{ tasks.resources }}"
+
+'''
+
+RETURN = '''
+entity:
+  description: Designated task is finished
+  returned: success
+  type: dict
+  contains:
+    task:
+        description: The finished task
+        type: dict
+'''
+
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import ForemanAnsibleModule
+
+
+class ForemanWaitForTask(ForemanAnsibleModule):
+    pass
+
+
+def main():
+    module = ForemanWaitForTask(
+        argument_spec=dict(
+            task=dict(type="str", required=True),
```

Comment:
I'm a bit out of touch, but a faint memory nags me that we may move this into `foreman_spec` and the lookup will be done for us. Is this correct?

Reply:
I'm pretty sure, `lookup_entity` works with the plain `ForemanAnsibleModule`. Or is the task not an entity in foreman is what you were trying to tell me?

https://github.com/theforeman/foreman-ansible-modules/blob/develop/plugins/module_utils/foreman_helper.py#L831

Reply:
Yeah. It's sort-of-an-entity-but-not-quite.

For example, it has no `name` (but `label` and `action`) attribute. And the only *unique* attribute is the `id` which we get passed here, but `lookup_entity` will, by default, try a search via name.
But given we already have the id, we don't need to perform a search at all (all it would give us is the same id anyway) and can just call `show_resource` on the ID directly (which we do in the code below).

Reply:
Right! Thank you.